File: 2019/py/day_05.py
from pathlib import Path

from intcode import CPU


def main():
    # Read in the program
    text = (Path(__file__).parent / "../input/input_05.txt").read_text()
    program = [int(x) for x in text.split(',')]
    # Run with each system ID
    for i, sys_id in enumerate([1, 5]):
        cpu = CPU(program.copy())
        cpu.input.append(sys_id)
        cpu.run()
        print(f"Part {i + 1}: {cpu.output[-1]}")


if __name__ == "__main__":
    main()

File: rust/2019/src/bin/day_05.rs
use std::fs;

use aoc_2019::CPU;

fn main() {
    let text = fs::read_to_string("../input/input_05.txt").unwrap();
    let program: Vec<_> = text.trim().split(',').map(|s| s.parse().unwrap()).collect();
    for (i, &p) in [1, 5].iter().enumerate() {
        let mut cpu = CPU::new(&program);
        cpu.input.push_back(p.into());
        cpu.run();
        println!("Part {}: {}", i + 1, cpu.output.pop_back().unwrap());
    }
}
